src/vm/internal/compiler/lex.cc

```diff
@@ -841,10 +846,82 @@ static int skip_to(const char *token, const char *atoken) {
   }
 }
 
+void init_include_path() {
+  svalue_t *ret;
```

Comment:
It's a shame that the younger generations have been taught that "goto" is
somehow evil.  Even in modern languages, it can often make MUCH easier to
understand code when you need to jump out of some deeply nested loop while
parsing something, and the alternative is cluttering things up with dozens
of empty else clauses, or even worse... code that's almost identical.

Yeah, I'm old, and I learned programming in BASIC and assembly first. :)

On Sun, Jan 17, 2016 at 12:40 PM, Shea690901 <notifications@github.com>
wrote:

> In src/vm/internal/compiler/lex.cc
> <https://github.com/fluffos/fluffos/pull/313#discussion_r49949392>:
>
> > @@ -841,10 +846,82 @@ static int skip_to(const char *token, const char *atoken) {
> >    }
> >  }
> >
> > +void init_include_path() {
> > +  svalue_t *ret;
>
> To use STL I had to rewrite the complete handling of include, something I
> don't want to try without being able to compile and run the driver on my
> own, although I have to agree that that is something to do in the future…
>
> Concerning goto:
> This (some cleanup in case of errors) is the only case I would allow the
> usage of goto, since the alternative of code doubling in my eyes is worse
> and yields easily to mistakes.
> The other alternative would be switching to classes, again something I
> won't try as long as I don't have a 'real' computer available to try those
> things on my own (my tablet doesn't even have a */bin/sh*! And I can't
> put one there either 'cause I don't have write permission on */*)…
>
> And for the readability I've added all those comments?!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fluffos/fluffos/pull/313/files#r49949392>.
>